### PR TITLE
Borrowings service

### PR DESCRIPTION
**Pull request by IvanStored**

### CODE REVIEW

**Comment by IvanStored** [comment]
i don`t sure about return borrowing

File: borrowings/models.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        default=calculate_expected_return_date(borrow_days=MAX_TERM)
+    )
+    actual_return_date = models.DateField(blank=True, null=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="borrowings",
+    )
+    book = models.ForeignKey(
+        Book, on_delete=models.CASCADE, related_name="borrowings"
```
Comment (zhulich) [inline borrowings/models.py:28]: i think here better use `PROTECT` so while user borrowing book we can't delete him and book

File: borrowings/serializers.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+class BorrowingDetailSerializer(BorrowingSerializer):
+
+    user = UserSerializer(many=False)
+    book = BookSerializer(many=False)
+
+
+class BorrowingReturnSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Borrowing
+        fields = ("actual_return_date",)
```
Comment (zhulich) [inline borrowings/serializers.py]: i think you have to override the field `actual_return_date` to current date

File: borrowings/models.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+from django.conf import settings
+from django.db import models
+from books.models import Book
+
+
+def calculate_expected_return_date(borrow_days: int) -> datetime:
+
+    current_date = datetime.date.today()
+
+    return current_date + datetime.timedelta(days=borrow_days)
```
Comment (zhulich) [inline borrowings/models.py:12]: it's necessary specify a variable `borrow_days ` if it's always the same number? you can solve it in one line

**Review by zhulich — CHANGES_REQUESTED** [review]

File: borrowings/views.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from borrowings.models import Borrowing
+from borrowings.serializers import (
+    BorrowingSerializer,
+    BorrowingListSerializer,
+    BorrowingDetailSerializer, BorrowingCreateSerializer,
+    BorrowingReturnSerializer
+)
```
Comment (sbublinskyi) [inline borrowings/views.py]: need to fix imports

File: borrowings/serializers.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    class Meta:
+        model = Borrowing
+        fields = (
+            "id",
+            "borrow_date",
+            "expected_return_date",
+            "actual_return_date",
+            "book_title",
+            "user_email",
+        )
```
Comment (sbublinskyi) [inline borrowings/serializers.py]: why to inherit from BorrowingSerializer, when you redefine everything in this class?

File: borrowings/views.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        if self.action == "retrieve":
+            return BorrowingDetailSerializer
+
+        if self.action == "create":
+            return BorrowingCreateSerializer
+        if self.action == "return_book":
+            return BorrowingReturnSerializer
+
+        return BorrowingSerializer
+
```
Comment (sbublinskyi) [inline borrowings/views.py]: With which action is expected to return BorrowingSerializer? 
I think, you don't need it at all

File: library_service/urls.py [hunk 7]
```diff
@@ -36,4 +36,5 @@
     ),
     path("api/", include("books.urls", namespace="books")),
     path("users/", include("user.urls", namespace="users")),
+    path("api/", include("borrowings.urls", namespace="borrowings")),
 ]
```
Comment (sbublinskyi) [inline library_service/urls.py]: I think, it is better to do pathes like this:
```
    path("api/books/", include("books.urls", namespace="books")),
    path("api/borrowings/", include("borrowings.urls", namespace="borrowings")),
    path("users/", include("user.urls", namespace="users")),

```

**Review by sbublinskyi — CHANGES_REQUESTED** [review]

**Review by sbublinskyi — APPROVED** [review]

File: borrowings/models.py (hunk 3 above)
Comment (IvanStored) [reply borrowings/models.py:12]: Yes, best practice :)

**Review by Kitazuka — APPROVED** [review]

**Pull request merged**